ci: add support for python 3.10 experimental builds

cereblanco opened the pull request:
# Pull Request Check List

Resolves: https://github.com/python-poetry/poetry/issues/3842

- Updated CIs for tests and skip.yaml
- I used `pip to install poetry` because Windows envs encounter error using `python install-poetry.py -y`

<!-- This is just a reminder about the most common mistakes. Please make sure that you tick all *appropriate* boxes.  But please read our [contribution guide](https://python-poetry.org/docs/contributing/) at least once, it will save you unnecessary review cycles! -->

~~[ ] Added **tests** for changed code.~~
~~[ ] Updated **documentation** for changed code.~~

<!-- If you have *any* questions to *any* of the points above, just **submit and ask**!  This checklist is here to *help* you, not to deter you from contributing! -->

abn commented on .github/workflows/main.yml:
```diff
@@ -18,11 +18,17 @@ on:
   tests:
     name: ${{ matrix.os }} / ${{ matrix.python-version }}
-    runs-on: ${{ matrix.os }}-latest
+    runs-on: ubuntu-latest
```
This should use `matrix.os` else we end up running all tests on linux.

abn commented on .github/workflows/main.yml:
```diff
@@ -39,7 +45,7 @@ jobs:
       - name: Bootstrap poetry
         shell: bash
-        run: python install-poetry.py -y
+        run: pip install poetry
```
This can remain using the install script.

abn commented on .github/workflows/skip.yml:
```diff
@@ -29,9 +29,15 @@ jobs:
     name: ${{ matrix.os }} / ${{ matrix.python-version }}
```
We shouldn't need changes here as we use this to trick the mandatory action checks. And this won't be mandatory until 3.10 is stable.

abn reviewed (changes requested):
Thank you @cereblanco , added some minor comments.

cereblanco commented:
> What was the error on using the install script? We should probably fix that.

@abn i reverted the unnecessary changes.. 
the error was caused by `runs-on: ubuntu-latest` instead of `matrix.os`  🤦🏼 😅 which i reverted as well

thank you
